tasks/main.yml

```diff
@@ -52,121 +49,55 @@
   copy:
     src: "{{ item.src }}"
     dest: "{{ item.dest }}"
-  with_items: "{{ automation_extra_files }}"
-  when: automation_extra_files is defined
-
-#
-# Install Replicated
-#
-# Parameters are:
-#   - private_address
-#   - public_address
-#   - proxy
-#   - docker_version
-#   - install replicated|operator; replicated is the default
-#   - install_url overwrite the default install url
-#
-
-- name: Set install parameters
+  with_items: "{{ replicated_automation_extra_files }}"
+  when: replicated_automation_extra_files is defined
+
+###
+### Install Replicated
+###
+### Parameters are:
+###   - replicated_private_address
+###   - replicated_public_address
+###   - replicated_proxy
+###   - replicated_docker_version
+###   - replicated_install replicated|operator; replicated is the default
+###
+
+###
+### Run the docker install for the operating system
+###
+
+- name: Check if Docker is already installed
+  stat:
+    path: /usr/bin/dockerd
+  register: docker_installed
+
+- name: Install Docker
+  include: docker/pre/{{ ansible_distribution }}.yml
+  when: docker_installed.stat.exists == False
```

Comment:
I don't think you should check if docker is installed here because the docker install book will also ensure that the correct version is installed.  What you should do here is add a variable similar to "no-docker" so that people can skip docker install if they want to.